frontend/src/utilities/segmentIOUtils.tsx

```diff
@@ -0,0 +1,91 @@
+import { TrackingEventProperties } from '../types';
+
+export const fireTrackingEvent = (
+  eventType: string,
+  properties?: TrackingEventProperties,
+): void => {
+  const clusterID = window.clusterID ?? '';
+  if (window.analytics) {
+    switch (eventType) {
+      case 'identify':
+        window.analytics.identify(properties?.anonymousID, { clusterID });
+        break;
+      case 'page':
+        window.analytics.page(undefined, { clusterID });
+        break;
+      default:
+        window.analytics.track(eventType, { ...properties, clusterID });
+    }
+  }
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const initSegment = async (props) => {
+  const { segmentKey, username } = props;
+  const analytics = (window.analytics = window.analytics || []);
```

Comment:
```suggestion
  const analytics = (window.analytics = window.analytics || {});
```

Reply:
It has to be an array, but not an object. I don't know why but it seems this is how the segment.io script was designed.